server/server.go

```diff
@@ -731,6 +731,17 @@ func (s *Server) checkAuth(c *client) bool {
 	}
 }
 
+// Check that number of clients is below Max connection setting.
+func (s *Server) checkMaxConn(c *client) bool {
+	if c.typ == CLIENT {
```

Comment:
And there is a huge defect here that I should check that s.opts.MaxConn is > 0 first!!!